src/deepsparse/transformers/pipelines/text_classification.py

```diff
@@ -62,8 +63,51 @@ class TextClassificationInput(BaseModel):
         "text_classification task"
     )
 
+    @staticmethod
+    def create_test_inputs(
+        batch_size: int = 1,
+    ) -> Dict[str, Union[List[List[str]], List[str], str]]:
+        """
+        Utility method to return a Dict representing sample inputs of given batch
+        size for current Schema
+        """
+        assert (
+            isinstance(batch_size, int) and batch_size > 0
+        ), "batch size must be greater than 1"
+        _inputs = ["I am Batman" for _ in range(batch_size)]
+        return {"sequences": _inputs}
+
+    def split(self) -> Generator["TextClassificationInput", None, None]:
+        """
+        Split a current `TextClassificationInput` object with a batch size b, into a
+        generator of b smaller objects with batch size 1, the returned
+        object can be iterated on.
+
+        :return: A Generator of smaller `TextClassificationInput` objects each
+            representing an input of batch-size 1
+        """
+
+        sequences = self.sequences
+
+        # case 1: do nothing if single input of batch_size 1
+        if isinstance(sequences, str):
+            yield self
+
+        elif (
+            isinstance(sequences, list)
+            and len(sequences)
+            and isinstance(sequences[0], (str, list))
+        ):
+            # case 2: List[str] -> multi-batches of size 1 Or batch-size 1 multi-inputs
+            # case 3: List[List[str]] -> Each List[str] is a multi-input batch of
+            # size 1
+            for sequence in sequences:
+                yield TextClassificationInput(sequences=sequence)
```

Comment:
not sure if this is correct. as discussed, we should be wrapping the broken list elements back into lists in most cases, right?